Dockerfile

```diff
@@ -1,29 +1,30 @@
-FROM edxops/xenial-common:latest
+FROM edxops/focal-common:latest
 RUN apt-get update && apt-get install -y \
     gettext \
     lib32z1-dev \
     libjpeg62-dev \
-    libxslt-dev \
-    libz-dev \
-    python3.5 \
+    libxslt1-dev \
+    zlib1g-dev \
+    python3 \
     python3-dev \
     python3-pip && \
     pip3 install --upgrade pip setuptools && \
     rm -rf /var/lib/apt/lists/*
 
-
 COPY . /usr/local/src/xblock-sdk
 WORKDIR /usr/local/src/xblock-sdk
 
-RUN pip install -r /usr/local/src/xblock-sdk/requirements/dev.txt 
+RUN ln -s /usr/bin/python3.8 /usr/bin/python && \
+    /usr/bin/python3 -m pip install --upgrade pip && \
+    pip install -r /usr/local/src/xblock-sdk/requirements/dev.txt
 
-RUN curl -sL https://deb.nodesource.com/setup_10.x -o /tmp/nodejs-setup
-RUN /bin/bash /tmp/nodejs-setup
-RUN rm /tmp/nodejs-setup
-RUN apt-get -y install nodejs
-RUN echo $PYTHONPATH
+RUN curl -sL https://deb.nodesource.com/setup_14.x -o /tmp/nodejs-setup && \
+    /bin/bash /tmp/nodejs-setup && \
+    rm /tmp/nodejs-setup && \
+    apt-get -y install nodejs && \
+    echo $PYTHONPATH && \
+    make install
 
-RUN make install
 EXPOSE 8000
```

Comment:
I already have some container using `8000` so wasn't able to test this. 🙂  Since `8000` is the default port for Django maybe we can change this to something which has less chance of a conflict, like `18340`?

Reply:
XBlock SDK uses port 8000 by default. I'm not sure I should be changing this.
I'll look at all the other comments more carefully next week.

Reply:
I'm not changing the port. I don't feel qualified to determine if such a change is appropriate.
Since the xblock-sdk was intended to provide a minimal framework for XBlock development, it was not intended to be a piece which integrates with a full dev-stack, but as an alternative.
Prior developers would be more qualified to make the call on whether changing the port number.

Reply:
Hmm, when I try to run `./manage runserver` inside the container I am still getting `Error: That port is already in use.`. How are you working with this?

Reply:
The XBlock SDK is meant to be used for development without a running devstack, etc.
Since that is the intended usage - the conflict is not supposed to occur.
I'm new to this, and am not taking the initiative to change to port number, someone else should consider if that is suitable and handle it in a different PR.

Reply:
Hmm, actually I tested this after restarting docker (so nothing else was running) which is why I asked. Let me look into this some more to see why it is not working for me.

Reply:
Sorry. I did not explain well.

As soon as I run `make dev.up` the expected server is available on `http://localhost:8000/`.

---

If I enter a shell using `make app-shell` and in the shell run
```
./manage.py runserver
```
I also get an error - as it is already running.
```
root@localhost:/usr/local/src/xblock-sdk# ./manage.py runserver
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2021 - 17:16:01
Django version 2.2.14, using settings 'workbench.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
Error: That port is already in use.

```

Reply:
Ah, I missed the `CMD` specified below. Great, it is working as expected!